kyo-offheap/native/src/main/scala/kyo/offheap/Memory.scala

```diff
@@ -0,0 +1,228 @@
+// kyo-offheap/native/src/main/scala/kyo/Memory.scala
+package kyo.offheap
+
+import scala.scalanative.unsafe._
+import scala.scalanative.libc.stdlib._
+import scala.annotation.implicitNotFound
+import scala.annotation.tailrec
+import scala.deriving.Mirror
+
+/** Memory provides a safe, effect-tracked interface for off-heap memory management for Scala Native. */
+opaque type Memory[A] = (Ptr[Byte], Long) // (pointer, element count)
```

Comment:
As mentioned in https://github.com/getkyo/kyo/issues/1053, `Memory` shouldn't be replicated. The file should be moved to `shared/src` and then `native/src` should provides stub implementations for Java's `MemorySegment` and `Arena`.